File: BOJ/220814/1244.py
switch_N = int(input())
switch_line = [0] + list(map(int,input().split()))
student_N = int(input())
for _ in range(student_N):
    # 남 1 여 2
    gen, num = map(int,input().split())
    if gen-1: # 여
        cnt = 1
        switch_line[num] ^= 1
        while True:
            if num + cnt > switch_N or num - cnt < 1:
                break
            if switch_line[num+cnt] == switch_line[num-cnt]:
                switch_line[num+cnt] ^= 1
                switch_line[num-cnt] ^= 1
            else:
                break
            cnt += 1

    else: # 남
        for i in range(num,switch_N+1,num):
            switch_line[i] ^= 1
switch_line = switch_line[1:]
i = 0
while True:
    print_line = switch_line[i:i+20]
    print(*print_line)
    i += 20
    if i >= len(switch_line):
        break

File: BOJ/250707/1244.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException{
        Scanner sc = new Scanner(System.in);
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int switchN = Integer.parseInt(br.readLine());
        List<Integer> switchList = new ArrayList<>();
        switchList.add(0);
        Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).forEach(switchList::add);
        int studentN = Integer.parseInt(br.readLine());
        for (int i=0; i<studentN; i++) {
            int[] gen_num = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
            int gen = gen_num[0];
            int num = gen_num[1];
            if (gen-1 > 0) {
                int cnt = 1;
                switchList.set(num, switchList.get(num)^1);
                while (true) {
                    if ((num + cnt) > switchN || (num-cnt) < 1) {
                        break;
                    }
                    if (switchList.get(num+cnt) == switchList.get(num-cnt)) {
                        switchList.set(num+cnt, switchList.get(num+cnt)^1);
                        switchList.set(num-cnt, switchList.get(num-cnt)^1);
                    } else {
                        break;
                    }
                    cnt += 1;
                }
            } else {
                for (int j=num;j<switchN+1;j += num) {
                    switchList.set(j, switchList.get(j)^1);
                }
            }
        }
        List<Integer> newList = new ArrayList<>(switchList.subList(1, switchList.size()));
        int i = 0;
        while (i < newList.size()) {
            int end = Math.min(i + 20, newList.size());
            List<Integer> printLine = newList.subList(i, end);
            stringPrinter(printLine);
            i += 20;
        }
    }
    public static <T> void stringPrinter(List<T> list) {
        StringJoiner joiner = new StringJoiner(" ");
        for (T n : list) {
            joiner.add(n.toString());
        }
        System.out.println(joiner.toString());
    }
}